MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -3506,31 +3515,81 @@ GetDxeMemoryProtectionSettings (
   return NULL;
 }
 
+/**
+  Uninstalls the Memory Attribute Protocol from all handles.
+**/
+VOID
+EFIAPI
+UninstallMemoryAttributeProtocol (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+  UINTN       HandleCount;
+  UINTN       Index;
+  EFI_HANDLE  *HandleBuffer;
+
+  if (MemoryAttributeProtocol == NULL) {
+    Status = gBS->LocateProtocol (&gEfiMemoryAttributeProtocolGuid, NULL, (VOID **)&MemoryAttributeProtocol);
+    if (EFI_ERROR (Status)) {
+      return;
+    }
+  }
+
+  Status = gBS->LocateHandleBuffer (
+                  ByProtocol,
+                  &gEfiMemoryAttributeProtocolGuid,
+                  NULL,
+                  &HandleCount,
+                  &HandleBuffer
+                  );
+
+  if (!EFI_ERROR (Status)) {
+    for (Index = 0; Index < HandleCount; Index++) {
+      Status = gBS->UninstallProtocolInterface (
+                      HandleBuffer[Index],
+                      &gEfiMemoryAttributeProtocolGuid,
+                      MemoryAttributeProtocol
+                      );
+      ASSERT_EFI_ERROR (Status);
+    }
+  }
+
+  if (HandleBuffer != NULL) {
+    FreePool (HandleBuffer);
+  }
+}
+
 /**
   Sets the NX compatibility global to FALSE so future checks to
-  IsSystemNxCompatible() will return FALSE.
+  IsEnhancedMemoryProtectionActive() will return FALSE.
 **/
 VOID
 EFIAPI
-TurnOffNxCompatibility (
+ActivateCompatibilityMode (
   VOID
   )
 {
-  if (mIsSystemNxCompatible) {
-    DEBUG ((DEBUG_INFO, "%a - Setting Nx on Code types to FALSE\n", __FUNCTION__));
+  if (!mEnhancedMemoryProtectionActive) {
+    return;
   }
 
-  mIsSystemNxCompatible = FALSE;
+  DEBUG ((DEBUG_INFO, "%a - Activating Memory Protection Compatibility Mode!\n", __FUNCTION__));

```

Comment:
```suggestion
  DEBUG ((DEBUG_ERROR, "%a - Activating Memory Protection Compatibility Mode!\n", __FUNCTION__));

```

To me this is an error condition (or at least worthy of an error message)

Reply:
Updated to ERROR